docs/gitrepo-structure.md

```diff
@@ -72,6 +72,10 @@ helm:
   # All labels on Rancher clusters are available using global.fleet.clusterLabels.LABELNAME
   # These can now be accessed directly as variables
     variableName: global.fleet.clusterLabels.LABELNAME
+  # They can also be access using go template logic
+    variableName: "kubernetes.io/cluster/{{ .global.fleet.clusterLabels.LABELNAME }}"
+    variableName: "{{ if eq .global.fleet.clusterLabels.LABELNAME \"production\" }}Production Workload{{ else }}Non Prod{{ end }}"
```

Comment:
That is an excellent point. I would think we could make this behavior opt in. So something like add parseValuesValues: true (that's a bad name fyi).  Then if somebody wants to templatize something that has templates in it they would escape the template characters.

Reply:
Yes, that is true.  An early version of the code actually would leave any lines alone that returned an error from the go templating call allowing fleet to still handle values files with go templating strings in them.  This however would not return errors to the user if they did something like this `variableName: "kubernetes.io/cluster/{{ .global.fleet.clusterLabels.MISSING_LABEL }}"` so I changed it as I forgot why I did it in the first place till you mentioned it.


Assuming I leave all clusterLabels prefixed with .global.fleet. a simple approach would be to only call to go templating if the string contained `{{`, `}}` and `.global.fleet.` which should in theory be safe. for most use cases.   Or only support the templating on stuff in fleet.yaml and not values files.

Reply:
I was thinking of `helm.preProcessValues` as the setting name to turn the templating on.  I actually submitted a patch to @rajiteh for his PR #671 instead of more updates to my code.  Granted I do have all the changes requested for this PR sitting locally but I think his approach is better you are should all review his PR.